docs/Migrating-Configuration.md

```diff
@@ -0,0 +1,27 @@
+While OTP's GraphQL APIs are very, very stable even across versions, the JSON configuration schema
+is not. Changes to it are relatively frequent and you can see all PRs that change it with
+the [Github tag 'config change'](https://github.com/opentripplanner/OpenTripPlanner/pulls?q=label%3A%22config+change%22).
+
+### Migrating the JSON configuration
+
+OTP validates the configuration and prints warnings during startup. This means that when you
+migrate to a newer version, you should carefully inspect the logs. If you see messages like
+
+```
+(NodeAdapter.java:170) Unexpected config parameter: 'routingDefaults.stairsReluctance:1.65' in 'router-config.json'
+```
+
+this means there are properties in your configuration that are unknown to OTP and therefore likely
+to be a configuration error, perhaps because the schema was changed. In such a case you should
+consult the [feature](Configuration.md#otp-features), [router](RouterConfiguration.md) or 
+[build configuration documentation](BuildConfiguration.md) to find out what the new schema looks like.
+
+### Aborting on invalid configuration
+
+If you want OTP to abort the startup when encountering invalid configuration, you can add the flag
+`--configCheck` to your regular OTP CLI commands.
+
+This should of course be used wisely as it can also accidentally make your production instances refuse 
+to start up.
+Therefore, we recommend that you use it only in a separate pre-production step, perhaps during graph
+build.
```

Comment:
The main model needs to be described here. 

OTP does not abort by default because:
 - I advanced deployments the configuration should follow the life-cycle of the environment.
 - Old config parameters should be left as is for a migration period. This allow you to role back OTP without the need to roll-back the OTP configuration. Let say you change the location of the graphs, then a critical error occours and you need to roll back. Any member of the dev-ops team should then be confident that she/he can roll back without risk - even if the environment changed.
 
 Note! We could improve this by separating environment spesiffic config and OTP turning.

Reply:
I took a stab at this here: 12d3d2daecd227d823f432bf5fe4c3296f0b93be

Let's have a quick chat in the meeting what exactly you would like to see there.